apps/client/src/features/table/store/CuesheetSettings.tsx

```diff
@@ -0,0 +1,55 @@
+import create from 'zustand';
+
+import { booleanFromLocalStorage } from '../../../common/utils/localStorage';
+
+interface CuesheetSettings {
+  showSettings: boolean;
+  followSelected: boolean;
+  showDelayBlock: boolean;
+  showPrevious: boolean;
+
+  toggleSettings: (val?: boolean) => void;
+  toggleFollow: (val?: boolean) => void;
+  toggleDelayVisibility: (val?: boolean) => void;
+  togglePreviousVisibility: (val?: boolean) => void;
+}
+
+function toggle(oldValue: boolean, value?: boolean) {
+  if (typeof value === 'undefined') {
+    return !oldValue;
+  }
+  return value;
+}
+
+enum CuesheetKeys {
+  Follow = 'ontime-cuesheet-follow-selected',
+  DelayVisibility = 'ontime-cuesheet-show-delay',
+  PreviousVisibility = 'ontime-cuesheet-show-previous',
+}
+
+export const useCuesheetSettings = create<CuesheetSettings>()((set) => ({
+  showSettings: false,
+  followSelected: booleanFromLocalStorage(CuesheetKeys.Follow, false),
+  showDelayBlock: booleanFromLocalStorage(CuesheetKeys.DelayVisibility, true),
+  showPrevious: booleanFromLocalStorage(CuesheetKeys.PreviousVisibility, true),
```

Comment:
Have you tried using the power of closures to use the `useLocalStorage` hook?

```ts
export const useCuesheetSettings = () => {

const [followedSelected, setFollowedSelected] = useLocalStorage(CuesheetKeys.Follow, false);
const [showDelayBlock, setShowDelayBlock] = useLocalStorage(CuesheetKeys.DelayVisibility, false);
const [showPrevious, setShowPrevious] = useLocalStorage(CuesheetKeys.PreviousVisibility, false);

return create<CuesheetSettings>()((set) => ({
  showSettings: false,
  followSelected,
  showDelayBlock,
  showPrevious,
  
  // in the setter functions below, 
  // rather than using `localStorage.setItem`, 
  // use the `setVariable` returned by the `useLocalStorage` hooks
  })
}
```